cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/analysis/fsm/DFAOrderEvaluator.kt

```diff
@@ -151,10 +167,39 @@ open class DFAOrderEvaluator(
             // try to make the transition in the DFA and retrieve the next nodes
             // to process for each of the paths.
             for (eogPath in eogPathSet) {
+                // Handle 'reset nodes'.
+                if (node in consideredResetNodes) {
+                    val baseAndOp =
+                        getBaseAndOpOfNode(node as CallExpression, eogPath, interproceduralFlows)
+                    if (baseAndOp != null) {
+                        val (base, _) = baseAndOp
+                        // get the DFA associated with this base
+                        val dfa = baseToFSM.computeIfAbsent(base) { dfa.deepCopy() }
+
+                        if (dfa.isAccepted) {
```

Comment:
A small comment would be good here so that we remember what we did ;)

Reply:
added a comment explaining that this evaluates the DFA when a reset node is encountered